<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Generate a HTML/PDF report exploring a set of genomic regions — renderReport • regionReport</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Generate a HTML/PDF report exploring a set of genomic regions — renderReport" />
<meta property="og:description" content="This function generates a HTML report with quality checks, genome location
exploration, and an interactive table with the results. Other output formats
are possible such as PDF but lose the interactivity. Users can easily append
to the report by providing a R Markdown file to customCode, or can
customize the entire template by providing an R Markdown file to
template." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">regionReport</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.25.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/regionReport.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/bumphunterExample.html">Example report using bumphunter results</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/leekgroup/regionReport/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Generate a HTML/PDF report exploring a set of genomic regions</h1>
    <small class="dont-index">Source: <a href='https://github.com/leekgroup/regionReport/blob/master/R/renderReport.R'><code>R/renderReport.R</code></a></small>
    <div class="hidden name"><code>renderReport.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function generates a HTML report with quality checks, genome location
exploration, and an interactive table with the results. Other output formats
are possible such as PDF but lose the interactivity. Users can easily append
to the report by providing a R Markdown file to <code>customCode</code>, or can
customize the entire template by providing an R Markdown file to
<code>template</code>.</p>
    </div>

    <pre class="usage"><span class='fu'>renderReport</span><span class='op'>(</span>
  <span class='va'>regions</span>,
  project <span class='op'>=</span> <span class='st'>""</span>,
  pvalueVars <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span>`P-values` <span class='op'>=</span> <span class='st'>"pval"</span><span class='op'>)</span>,
  densityVars <span class='op'>=</span> <span class='cn'>NULL</span>,
  significantVar <span class='op'>=</span> <span class='fu'>mcols</span><span class='op'>(</span><span class='va'>regions</span><span class='op'>)</span><span class='op'>$</span><span class='va'>pval</span> <span class='op'>&lt;=</span> <span class='fl'>0.05</span>,
  annotation <span class='op'>=</span> <span class='cn'>NULL</span>,
  nBestRegions <span class='op'>=</span> <span class='fl'>500</span>,
  customCode <span class='op'>=</span> <span class='cn'>NULL</span>,
  outdir <span class='op'>=</span> <span class='st'>"regionExploration"</span>,
  output <span class='op'>=</span> <span class='st'>"regionExploration"</span>,
  browse <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/interactive.html'>interactive</a></span><span class='op'>(</span><span class='op'>)</span>,
  txdb <span class='op'>=</span> <span class='cn'>NULL</span>,
  device <span class='op'>=</span> <span class='st'>"png"</span>,
  densityTemplates <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>Pvalue <span class='op'>=</span> <span class='va'>templatePvalueDensity</span>, Common <span class='op'>=</span> <span class='va'>templateDensity</span>,
    Manhattan <span class='op'>=</span> <span class='va'>templateManhattan</span><span class='op'>)</span>,
  template <span class='op'>=</span> <span class='cn'>NULL</span>,
  theme <span class='op'>=</span> <span class='cn'>NULL</span>,
  digits <span class='op'>=</span> <span class='fl'>2</span>,
  <span class='va'>...</span>
<span class='op'>)</span>

<span class='va'>templatePvalueDensity</span>

<span class='va'>templateDensity</span>

<span class='va'>templateManhattan</span>

<span class='va'>templatePvalueHistogram</span>

<span class='va'>templateHistogram</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>regions</th>
      <td><p>The set of genomic regions of interest as a <code>GRanges</code>
object. All sequence lengths must be provided.</p></td>
    </tr>
    <tr>
      <th>project</th>
      <td><p>The title of the project.</p></td>
    </tr>
    <tr>
      <th>pvalueVars</th>
      <td><p>The names of the variables with values between 0 and 1 to
plot density values by chromosome and a table for commonly used cutoffs.
Most commonly used to explore p-value distributions. If a named
character vector is provided, the names are used in the plot titles.</p></td>
    </tr>
    <tr>
      <th>densityVars</th>
      <td><p>The names of variables to use for making density plots
by chromosome. Commonly used to explore scores and other variables given
by region.  If a named character vector is provided, the names are used in
the plot titles.</p></td>
    </tr>
    <tr>
      <th>significantVar</th>
      <td><p>A <code>logical</code> variable differentiating statistically
significant regions from the rest. When provided, both types of regions
are compared against each other to see differences in width, location, etc.</p></td>
    </tr>
    <tr>
      <th>annotation</th>
      <td><p>The output from <a href='https://rdrr.io/pkg/bumphunter/man/matchGenes.html'>matchGenes</a> used on
<code>regions</code>. Note that this can take time for a large set of regions
so it's better to pre-compute this information and save it.</p></td>
    </tr>
    <tr>
      <th>nBestRegions</th>
      <td><p>The number of regions to include in the interactive
table.</p></td>
    </tr>
    <tr>
      <th>customCode</th>
      <td><p>An absolute path to a child R Markdown file with code to be
evaluated before the reproducibility section. Its useful for users who want
to customize the report by adding conclusions derived from the data and/or
further quality checks and plots.</p></td>
    </tr>
    <tr>
      <th>outdir</th>
      <td><p>The name of output directory.</p></td>
    </tr>
    <tr>
      <th>output</th>
      <td><p>The name of output HTML file (without the html extension).</p></td>
    </tr>
    <tr>
      <th>browse</th>
      <td><p>If <code>TRUE</code> the HTML report is opened in your browser once
it's completed.</p></td>
    </tr>
    <tr>
      <th>txdb</th>
      <td><p>Specify the transcription database to use for identifying the
closest genes via <a href='https://rdrr.io/pkg/bumphunter/man/matchGenes.html'>matchGenes</a>. If <code>NULL</code> it will
use TxDb.Hsapiens.UCSC.hg19.knownGene by default.</p></td>
    </tr>
    <tr>
      <th>device</th>
      <td><p>The graphical device used when knitting. See more at
<a href='http://yihui.name/knitr/options'>http://yihui.name/knitr/options</a> (<code>dev</code> argument).</p></td>
    </tr>
    <tr>
      <th>densityTemplates</th>
      <td><p>A list of length 3 with templates for the p-value
density plots (variables from <code>pvalueVars</code>), the continuous
variables density plots (variables from <code>densityVars</code>), and Manhattan
plots for the p-value variables (<code>pvalueVars</code>). These templates
are processed by <a href='https://rdrr.io/pkg/whisker/man/whisker.render.html'>whisker.render</a>. Check the default templates
for more information. The <code>densityTemplates</code> argument is available for
those users interested in customizing these plots. For example, to show
histograms instead of density plots use <code>templatePvalueHistogram</code> and
<code>templateHistogram</code> instead of <code>templatePvalueDensity</code> and
<code>templateDensity</code> respectively.</p></td>
    </tr>
    <tr>
      <th>template</th>
      <td><p>Template file to use for the report. If not provided, will
use the default file found in regionExploration/regionExploration.Rmd
within the package source.</p></td>
    </tr>
    <tr>
      <th>theme</th>
      <td><p>A ggplot2 <a href='https://ggplot2.tidyverse.org/reference/theme.html'>theme</a> to use for the plots made with
ggplot2.</p></td>
    </tr>
    <tr>
      <th>digits</th>
      <td><p>The number of digits to round to in the interactive table of
the top <code>nBestRegions</code>. Note that p-values and adjusted p-values won't
be rounded.</p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Arguments passed to other methods and/or advanced arguments.
Advanced arguments:</p><dl>
<dt>overviewParams </dt><dd><p>A two element list with <code>base_size</code> and
<code>areaRel</code> that control the text size for the genomic overview plots.</p></dd>
<dt>output_format </dt><dd><p>Either <code>html_document</code>, <code>pdf_document</code> or
<code><a href='https://rdrr.io/pkg/knitrBootstrap/man/bootstrap_document.html'>knitrBootstrap::bootstrap_document</a></code> unless you modify the YAML
template.</p></dd>
<dt>clean </dt><dd><p>Logical, whether to clean the results or not. Passed to
<a href='https://rdrr.io/pkg/rmarkdown/man/render.html'>render</a>.</p></dd>

</dl></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="format"><a class="anchor" href="#format"></a>Format</h2>

    <p>An object of class <code>character</code> of length 1.</p>
<p>An object of class <code>character</code> of length 1.</p>
<p>An object of class <code>character</code> of length 1.</p>
<p>An object of class <code>character</code> of length 1.</p>
<p>An object of class <code>character</code> of length 1.</p>
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>An HTML report with a basic exploration for the given set of
genomic regions. See the example report at
<a href='http://leekgroup.github.io/regionReport/reference/renderReport-example/regionExploration.html'>http://leekgroup.github.io/regionReport/reference/renderReport-example/regionExploration.html</a>.</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>Set <code>output_format</code> to <code>'knitrBootstrap::bootstrap_document'</code> or
<code>'pdf_document'</code> if you want a HTML report styled by knitrBootstrap or
a PDF report respectively. If using knitrBootstrap, we recommend the version
available only via GitHub at <a href='https://github.com/jimhester/knitrBootstrap'>https://github.com/jimhester/knitrBootstrap</a>
which has nicer features than the current version available via CRAN. You can
also set the <code>output_format</code> to <code>'html_document'</code> for a HTML
report styled by rmarkdown. The default is set to
<code>'BiocStyle::html_document'</code>.</p>
<p>If you modify the YAML front matter of <code>template</code>, you can use other
values for <code>output_format</code>.</p>
<p>The HTML report styled with knitrBootstrap can be smaller in size than the
<code>'html_document'</code> report.</p>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Leonardo Collado-Torres</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'>
<span class='co'>## Load derfinder for an example set of regions</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='st'><a href='https://github.com/lcolladotor/derfinder'>"derfinder"</a></span><span class='op'>)</span>
<span class='va'>regions</span> <span class='op'>&lt;-</span> <span class='va'>genomeRegions</span><span class='op'>$</span><span class='va'>regions</span>

<span class='co'>## Assign chr length</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='st'><a href='https://bioconductor.org/packages/GenomicRanges'>"GenomicRanges"</a></span><span class='op'>)</span>
<span class='fu'>seqlengths</span><span class='op'>(</span><span class='va'>regions</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"chr21"</span> <span class='op'>=</span> <span class='fl'>48129895</span><span class='op'>)</span>

<span class='co'>## The output will be saved in the 'renderReport-example' directory</span>
<span class='fu'><a href='https://rdrr.io/r/base/files2.html'>dir.create</a></span><span class='op'>(</span><span class='st'>"renderReport-example"</span>, showWarnings <span class='op'>=</span> <span class='cn'>FALSE</span>, recursive <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>

<span class='co'>## Generate the HTML report</span>
<span class='va'>report</span> <span class='op'>&lt;-</span> <span class='fu'>renderReport</span><span class='op'>(</span><span class='va'>regions</span>, <span class='st'>"Example run"</span>,
    pvalueVars <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span>
        <span class='st'>"Q-values"</span> <span class='op'>=</span> <span class='st'>"qvalues"</span>, <span class='st'>"P-values"</span> <span class='op'>=</span> <span class='st'>"pvalues"</span>
    <span class='op'>)</span>, densityVars <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span>
        <span class='st'>"Area"</span> <span class='op'>=</span> <span class='st'>"area"</span>, <span class='st'>"Mean coverage"</span> <span class='op'>=</span> <span class='st'>"meanCoverage"</span>
    <span class='op'>)</span>,
    significantVar <span class='op'>=</span> <span class='va'>regions</span><span class='op'>$</span><span class='va'>qvalues</span> <span class='op'>&lt;=</span> <span class='fl'>0.05</span>, nBestRegions <span class='op'>=</span> <span class='fl'>20</span>,
    outdir <span class='op'>=</span> <span class='st'>"renderReport-example"</span>
<span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Writing 11 Bibtex entries ... </span></div><div class='output co'>#&gt; <span class='message'>OK</span>
#&gt; <span class='message'>Results written to file 'renderReport-example/regionExploration.bib'</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>processing file: regionExploration.Rmd</span></div><div class='output co'>#&gt;   |                                                                                                                      |                                                                                                              |   0%  |                                                                                                                      |...                                                                                                           |   2%
#&gt;    inline R code fragments
#&gt; 
#&gt;   |                                                                                                                      |.....                                                                                                         |   5%
#&gt; label: docSetup (with options) 
#&gt; List of 3
#&gt;  $ bootstrap.show.code   : logi FALSE
#&gt;  $ dev                   : symbol device
#&gt;  $ bootstrap.show.message: logi FALSE
#&gt; 
#&gt;   |                                                                                                                      |........                                                                                                      |   7%
#&gt;   ordinary text without R code
#&gt; 
#&gt;   |                                                                                                                      |..........                                                                                                    |   9%
#&gt; label: setup (with options) 
#&gt; List of 1
#&gt;  $ bootstrap.show.message: logi FALSE
#&gt; 
#&gt;   |                                                                                                                      |.............                                                                                                 |  12%
#&gt;   ordinary text without R code
#&gt; 
#&gt;   |                                                                                                                      |...............                                                                                               |  14%
#&gt; label: pvaluePlots (with options) 
#&gt; List of 3
#&gt;  $ echo   : logi FALSE
#&gt;  $ results: chr "asis"
#&gt;  $ eval   : symbol hasPvalueVars
#&gt; </div><div class='output co'>#&gt;   |                                                                                                                      |..................                                                                                            |  16%
#&gt;   ordinary text without R code
#&gt; 
#&gt;   |                                                                                                                      |....................                                                                                          |  19%
#&gt; label: regLen (with options) 
#&gt; List of 4
#&gt;  $ fig.width : num 14
#&gt;  $ fig.height: num 14
#&gt;  $ eval      : symbol hasSignificant
#&gt;  $ echo      : symbol hasSignificant
#&gt; </div><div class='output co'>#&gt;   |                                                                                                                      |.......................                                                                                       |  21%
#&gt;   ordinary text without R code
#&gt; 
#&gt;   |                                                                                                                      |..........................                                                                                    |  23%
#&gt; label: regLen2 (with options) 
#&gt; List of 4
#&gt;  $ fig.width : num 10
#&gt;  $ fig.height: num 10
#&gt;  $ eval      : language !hasSignificant
#&gt;  $ echo      : language !hasSignificant
#&gt; 
#&gt;   |                                                                                                                      |............................                                                                                  |  26%
#&gt;    inline R code fragments
#&gt; 
#&gt;   |                                                                                                                      |...............................                                                                               |  28%
#&gt; label: densityPlots (with options) 
#&gt; List of 3
#&gt;  $ echo   : logi FALSE
#&gt;  $ results: chr "asis"
#&gt;  $ eval   : symbol hasDensityVars
#&gt; </div><div class='output co'>#&gt;   |                                                                                                                      |.................................                                                                             |  30%
#&gt;    inline R code fragments
#&gt; 
#&gt;   |                                                                                                                      |....................................                                                                          |  33%
#&gt; label: genomeOverview1 (with options) 
#&gt; List of 6
#&gt;  $ message   : logi FALSE
#&gt;  $ fig.width : num 7
#&gt;  $ fig.height: num 9
#&gt;  $ dpi       : num 300
#&gt;  $ eval      : symbol hasSignificant
#&gt;  $ echo      : symbol hasSignificant
#&gt; </div><div class='output co'>#&gt; <span class='message'>Scale for 'x' is already present. Adding another scale for 'x', which will replace the existing scale.</span></div><div class='output co'>#&gt; <span class='message'>Scale for 'x' is already present. Adding another scale for 'x', which will replace the existing scale.</span></div><div class='output co'>#&gt;   |                                                                                                                      |......................................                                                                        |  35%
#&gt;    inline R code fragments
#&gt; 
#&gt;   |                                                                                                                      |.........................................                                                                     |  37%
#&gt; label: manhattanPlots (with options) 
#&gt; List of 3
#&gt;  $ echo   : logi FALSE
#&gt;  $ results: chr "asis"
#&gt;  $ eval   : symbol hasPvalueVars
#&gt; </div><div class='output co'>#&gt;   |                                                                                                                      |...........................................                                                                   |  40%
#&gt;   ordinary text without R code
#&gt; 
#&gt;   |                                                                                                                      |..............................................                                                                |  42%
#&gt; label: genomeOverview2 (with options) 
#&gt; List of 4
#&gt;  $ message   : logi FALSE
#&gt;  $ fig.width : num 7
#&gt;  $ fig.height: num 9
#&gt;  $ dpi       : num 300
#&gt; </div><div class='output co'>#&gt; <span class='message'>No annotationPackage supplied. Trying org.Hs.eg.db.</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: org.Hs.eg.db</span></div><div class='output co'>#&gt; <span class='message'>Could not load org.Hs.eg.db. Will continue without annotation</span></div><div class='output co'>#&gt; <span class='message'>Getting TSS and TSE.</span></div><div class='output co'>#&gt; <span class='message'>Getting CSS and CSE.</span></div><div class='output co'>#&gt; <span class='message'>Getting exons.</span></div><div class='output co'>#&gt; <span class='message'>Scale for 'x' is already present. Adding another scale for 'x', which will replace the existing scale.</span></div><div class='output co'>#&gt;   |                                                                                                                      |.................................................                                                             |  44%
#&gt;    inline R code fragments
#&gt; 
#&gt;   |                                                                                                                      |...................................................                                                           |  47%
#&gt; label: annoReg (with options) 
#&gt; List of 1
#&gt;  $ results: chr "asis"
#&gt; 
#&gt;   |                                                                                                                      |......................................................                                                        |  49%
#&gt;   ordinary text without R code
#&gt; 
#&gt;   |                                                                                                                      |........................................................                                                      |  51%
#&gt; label: genomeOverview3 (with options) 
#&gt; List of 6
#&gt;  $ message   : logi FALSE
#&gt;  $ fig.width : num 7
#&gt;  $ fig.height: num 9
#&gt;  $ dpi       : num 300
#&gt;  $ eval      : symbol hasSignificant
#&gt;  $ echo      : symbol hasSignificant
#&gt; </div><div class='output co'>#&gt; <span class='message'>Scale for 'x' is already present. Adding another scale for 'x', which will replace the existing scale.</span></div><div class='output co'>#&gt;   |                                                                                                                      |...........................................................                                                   |  53%
#&gt;    inline R code fragments
#&gt; 
#&gt;   |                                                                                                                      |.............................................................                                                 |  56%
#&gt; label: countTable (with options) 
#&gt; List of 1
#&gt;  $ results: chr "asis"
#&gt; 
#&gt;   |                                                                                                                      |................................................................                                              |  58%
#&gt;   ordinary text without R code
#&gt; 
#&gt;   |                                                                                                                      |...................................................................                                           |  60%
#&gt; label: vennDiagram (with options) 
#&gt; List of 2
#&gt;  $ fig.width : num 7
#&gt;  $ fig.height: num 7
#&gt; </div><div class='output co'>#&gt;   |                                                                                                                      |.....................................................................                                         |  63%
#&gt;    inline R code fragments
#&gt; 
#&gt;   |                                                                                                                      |........................................................................                                      |  65%
#&gt; label: vennDiagramSignificant (with options) 
#&gt; List of 4
#&gt;  $ eval      : symbol hasSignificant
#&gt;  $ echo      : symbol hasSignificant
#&gt;  $ fig.width : num 7
#&gt;  $ fig.height: num 7
#&gt; </div><div class='output co'>#&gt;   |                                                                                                                      |..........................................................................                                    |  67%
#&gt;    inline R code fragments
#&gt; 
#&gt;   |                                                                                                                      |.............................................................................                                 |  70%
#&gt; label: bestRegionInfo (with options) 
#&gt; List of 1
#&gt;  $ results: chr "asis"
#&gt; 
#&gt;   |                                                                                                                      |...............................................................................                               |  72%
#&gt;   ordinary text without R code
#&gt; 
#&gt;   |                                                                                                                      |..................................................................................                            |  74%
#&gt; label: unnamed-chunk-1 (with options) 
#&gt; List of 2
#&gt;  $ child: symbol customCode
#&gt;  $ eval : symbol hasCustomCode
#&gt; 
#&gt;   |                                                                                                                      |....................................................................................                          |  77%
#&gt;    inline R code fragments
#&gt; 
#&gt;   |                                                                                                                      |.......................................................................................                       |  79%
#&gt; label: thecall (with options) 
#&gt; List of 1
#&gt;  $ echo: logi FALSE
#&gt; 
#&gt;   |                                                                                                                      |..........................................................................................                    |  81%
#&gt;   ordinary text without R code
#&gt; 
#&gt;   |                                                                                                                      |............................................................................................                  |  84%
#&gt; label: reproducibility1 (with options) 
#&gt; List of 1
#&gt;  $ echo: logi FALSE
#&gt; 
#&gt;   |                                                                                                                      |...............................................................................................               |  86%
#&gt;   ordinary text without R code
#&gt; 
#&gt;   |                                                                                                                      |.................................................................................................             |  88%
#&gt; label: reproducibility2 (with options) 
#&gt; List of 1
#&gt;  $ echo: logi FALSE
#&gt; 
#&gt;   |                                                                                                                      |....................................................................................................          |  91%
#&gt;   ordinary text without R code
#&gt; 
#&gt;   |                                                                                                                      |......................................................................................................        |  93%
#&gt; label: reproducibility3 (with options) 
#&gt; List of 1
#&gt;  $ echo: logi FALSE
#&gt; 
#&gt;   |                                                                                                                      |.........................................................................................................     |  95%
#&gt;    inline R code fragments
#&gt; 
#&gt;   |                                                                                                                      |...........................................................................................................   |  98%
#&gt; label: bibliography (with options) 
#&gt; List of 3
#&gt;  $ results: chr "asis"
#&gt;  $ echo   : logi FALSE
#&gt;  $ warning: logi FALSE
#&gt; 
#&gt;   |                                                                                                                      |..............................................................................................................| 100%
#&gt;   ordinary text without R code
#&gt; 
#&gt; </div><div class='output co'>#&gt; <span class='message'>output file: regionExploration.knit.md</span></div><div class='output co'>#&gt; /usr/local/bin/pandoc +RTS -K512m -RTS regionExploration.utf8.md --to html4 --from markdown+autolink_bare_uris+tex_math_single_backslash --output regionExploration.html --lua-filter /__w/_temp/Library/bookdown/rmarkdown/lua/custom-environment.lua --lua-filter /__w/_temp/Library/rmarkdown/rmarkdown/lua/pagebreak.lua --lua-filter /__w/_temp/Library/rmarkdown/rmarkdown/lua/latex-div.lua --self-contained --wrap preserve --standalone --section-divs --table-of-contents --toc-depth 3 --variable toc_float=1 --variable toc_selectors=h1,h2,h3 --variable toc_collapsed=1 --variable toc_smooth_scroll=1 --variable toc_print=1 --template /tmp/Rtmpkhkn6V/BiocStyle/template.html --no-highlight --variable highlightjs=1 --number-sections --css /__w/_temp/Library/BiocStyle/resources/html/bioconductor.css --variable 'theme:bootstrap' --include-in-header /tmp/Rtmpkhkn6V/rmarkdown-str259833621622.html --mathjax --variable 'mathjax-url:https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML' --variable code_folding=hide --variable code_menu=1 </div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Output created: regionExploration.html</span></div><div class='input'>
<span class='kw'>if</span> <span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/interactive.html'>interactive</a></span><span class='op'>(</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>{</span>
    <span class='co'>## Browse the report</span>
    <span class='fu'><a href='https://rdrr.io/r/utils/browseURL.html'>browseURL</a></span><span class='op'>(</span><span class='va'>report</span><span class='op'>)</span>
<span class='op'>}</span>

<span class='co'>## See the example report at</span>
<span class='co'>## http://leekgroup.github.io/regionReport/reference/renderReport-example/regionExploration.html</span>


<span class='co'>## Check the default templates. For users interested in customizing these</span>
<span class='co'>## plots.</span>
<span class='co'>## For p-value variables:</span>
<span class='fu'><a href='https://rdrr.io/r/base/cat.html'>cat</a></span><span class='op'>(</span><span class='fu'>regionReport</span><span class='fu'>::</span><span class='va'>templatePvalueDensity</span><span class='op'>)</span>
</div><div class='output co'>#&gt; 
#&gt; ## {{{densityVarName}}}
#&gt; 
#&gt; ```{r pval-density-{{{varName}}}, fig.width=10, fig.height=10}
#&gt; p1{{{varName}}} &lt;- ggplot(regions.df.plot, aes(x={{{varName}}}, colour=seqnames)) +
#&gt;     geom_line(stat='density') + xlim(0, 1) +
#&gt;     labs(title='Density of {{{densityVarName}}}') + xlab('{{{densityVarName}}}') +
#&gt;     scale_colour_discrete(limits=chrs) + theme(legend.title=element_blank())
#&gt; p1{{{varName}}}
#&gt; ```
#&gt; 
#&gt; 
#&gt; ```{r 'pval-summary-{{{varName}}}'}
#&gt; summary(mcols(regions)[['{{{varName}}}']])
#&gt; ```
#&gt; 
#&gt; 
#&gt; This is the numerical summary of the distribution of the {{{densityVarName}}}.
#&gt; 
#&gt; ```{r pval-tableSummary-{{{varName}}}, results='asis'}
#&gt; {{{varName}}}table &lt;- lapply(c(1e-04, 0.001, 0.01, 0.025, 0.05, 0.1, 0.2, 0.3, 0.4, 0.5,
#&gt;     0.6, 0.7, 0.8, 0.9, 1), function(x) {
#&gt;     data.frame('Cut' = x, 'Count' = sum(mcols(regions)[['{{{varName}}}']] &lt;= x))
#&gt; })
#&gt; {{{varName}}}table &lt;- do.call(rbind, {{{varName}}}table)
#&gt; if(outputIsHTML) {
#&gt;     kable({{{varName}}}table, format = 'markdown', align = c('c', 'c'))
#&gt; } else {
#&gt;     kable({{{varName}}}table)
#&gt; }
#&gt; ```
#&gt; 
#&gt; This table shows the number of regions with {{{densityVarName}}} less or equal than some commonly used cutoff values.
#&gt; </div><div class='input'>
<span class='co'>## For continous variables:</span>
<span class='fu'><a href='https://rdrr.io/r/base/cat.html'>cat</a></span><span class='op'>(</span><span class='fu'>regionReport</span><span class='fu'>::</span><span class='va'>templateDensity</span><span class='op'>)</span>
</div><div class='output co'>#&gt; 
#&gt; ## {{{densityVarName}}}
#&gt; 
#&gt; ```{r density-{{{varName}}}, fig.width=14, fig.height=14, eval=hasSignificant, echo=hasSignificant}
#&gt; xrange &lt;- range(regions.df.plot[, '{{{varName}}}']) * c(0.95, 1.05)
#&gt; p3a{{{varName}}} &lt;- ggplot(regions.df.plot[is.finite(regions.df.plot[, '{{{varName}}}']), ], aes(x={{{varName}}}, colour=seqnames)) +
#&gt;     geom_line(stat='density') + labs(title='Density of {{{densityVarName}}}') +
#&gt;     xlab('{{{densityVarName}}}') + scale_colour_discrete(limits=chrs) +
#&gt;     xlim(xrange) + theme(legend.title=element_blank())
#&gt; p3b{{{varName}}} &lt;- ggplot(regions.df.sig[is.finite(regions.df.sig[, '{{{varName}}}']), ], aes(x={{{varName}}}, colour=seqnames)) +
#&gt;     geom_line(stat='density') +
#&gt;     labs(title='Density of {{{densityVarName}}} (significant only)') +
#&gt;     xlab('{{{densityVarName}}}') + scale_colour_discrete(limits=chrs) +
#&gt;     xlim(xrange) + theme(legend.title=element_blank())
#&gt; grid.arrange(p3a{{{varName}}}, p3b{{{varName}}})
#&gt; ```
#&gt; 
#&gt; ```{r density-solo-{{{varName}}}, fig.width=10, fig.height=10, eval=!hasSignificant, echo=!hasSignificant}
#&gt; p3a{{{varName}}} &lt;- ggplot(regions.df.plot[is.finite(regions.df.plot[, '{{{varName}}}']), ], aes(x={{{varName}}}, colour=seqnames)) +
#&gt;     geom_line(stat='density') + labs(title='Density of {{{densityVarName}}}') +
#&gt;     xlab('{{{densityVarName}}}') + scale_colour_discrete(limits=chrs) +
#&gt;     theme(legend.title=element_blank())
#&gt; p3a{{{varName}}}
#&gt; ```
#&gt; 
#&gt; This plot shows the density of the {{{densityVarName}}} for all regions. `r ifelse(hasSignificant, 'The bottom panel is restricted to significant regions.', '')`
#&gt; </div><div class='input'>
<span class='co'>## For Manhattan plots</span>
<span class='fu'><a href='https://rdrr.io/r/base/cat.html'>cat</a></span><span class='op'>(</span><span class='fu'>regionReport</span><span class='fu'>::</span><span class='va'>templateManhattan</span><span class='op'>)</span>
</div><div class='output co'>#&gt; 
#&gt; ## Manhattan {{{densityVarName}}}
#&gt; 
#&gt; ```{r manhattan-{{{varName}}}, fig.width=10, fig.height=10, message = FALSE}
#&gt; 
#&gt; regions.manhattan &lt;- regions
#&gt; mcols(regions.manhattan)[['{{{varName}}}']] &lt;- - log(mcols(regions.manhattan)[['{{{varName}}}']], base = 10)
#&gt; pMan{{{varName}}} &lt;- plotGrandLinear(regions.manhattan, aes(y = {{{varName}}}, colour = seqnames)) + theme(axis.text.x=element_text(angle=-90, hjust=0)) + ylab('-log10 {{{densityVarName}}}')
#&gt; pMan{{{varName}}}
#&gt; rm(regions.manhattan)
#&gt; ```
#&gt; 
#&gt; This is a Manhattan plot for the {{{densityVarName}}} for all regions. A single dot is shown for each region, where higher values in the y-axis mean that the {{{densityVarName}}} are closer to zero.
#&gt; </div><div class='input'>
<span class='co'>##################################################</span>
<span class='co'>## bumphunter example mentioned in the vignette ##</span>
<span class='co'>##################################################</span>

<span class='co'>## Load bumphunter</span>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='st'><a href='https://github.com/ririzarr/bumphunter'>"bumphunter"</a></span><span class='op'>)</span>

<span class='co'>## Create data from the vignette</span>
<span class='va'>pos</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>
    pos1 <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='fl'>1000</span>, <span class='fl'>35</span><span class='op'>)</span>,
    pos2 <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span><span class='op'>(</span><span class='fl'>2001</span>, <span class='fl'>3000</span>, <span class='fl'>35</span><span class='op'>)</span>,
    pos3 <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='fl'>1000</span>, <span class='fl'>50</span><span class='op'>)</span>
<span class='op'>)</span>
<span class='va'>chr</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/rep.html'>rep</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='st'>"chr"</span>, <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='fl'>1</span>, <span class='fl'>2</span><span class='op'>)</span><span class='op'>)</span>, times <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/lapply.html'>sapply</a></span><span class='op'>(</span><span class='va'>pos</span>, <span class='va'>length</span><span class='op'>)</span><span class='op'>)</span>
<span class='va'>pos</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/unlist.html'>unlist</a></span><span class='op'>(</span><span class='va'>pos</span>, use.names <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span>

<span class='co'>## Find clusters</span>
<span class='va'>cl</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/bumphunter/man/clusterMaker.html'>clusterMaker</a></span><span class='op'>(</span><span class='va'>chr</span>, <span class='va'>pos</span>, maxGap <span class='op'>=</span> <span class='fl'>300</span><span class='op'>)</span>

<span class='co'>## Build simulated bumps</span>
<span class='va'>Indexes</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/split.html'>split</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq_along</a></span><span class='op'>(</span><span class='va'>cl</span><span class='op'>)</span>, <span class='va'>cl</span><span class='op'>)</span>
<span class='va'>beta1</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/rep.html'>rep</a></span><span class='op'>(</span><span class='fl'>0</span>, <span class='fu'><a href='https://rdrr.io/r/base/length.html'>length</a></span><span class='op'>(</span><span class='va'>pos</span><span class='op'>)</span><span class='op'>)</span>
<span class='kw'>for</span> <span class='op'>(</span><span class='va'>i</span> <span class='kw'>in</span> <span class='fu'><a href='https://rdrr.io/r/base/seq.html'>seq</a></span><span class='op'>(</span>along <span class='op'>=</span> <span class='va'>Indexes</span><span class='op'>)</span><span class='op'>)</span> <span class='op'>{</span>
    <span class='va'>ind</span> <span class='op'>&lt;-</span> <span class='va'>Indexes</span><span class='op'>[[</span><span class='va'>i</span><span class='op'>]</span><span class='op'>]</span>
    <span class='va'>x</span> <span class='op'>&lt;-</span> <span class='va'>pos</span><span class='op'>[</span><span class='va'>ind</span><span class='op'>]</span>
    <span class='va'>z</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/scale.html'>scale</a></span><span class='op'>(</span><span class='va'>x</span>, <span class='fu'><a href='https://rdrr.io/r/stats/median.html'>median</a></span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span>, <span class='fu'><a href='https://rdrr.io/r/base/Extremes.html'>max</a></span><span class='op'>(</span><span class='va'>x</span><span class='op'>)</span> <span class='op'>/</span> <span class='fl'>12</span><span class='op'>)</span>
    <span class='va'>beta1</span><span class='op'>[</span><span class='va'>ind</span><span class='op'>]</span> <span class='op'>&lt;-</span> <span class='va'>i</span> <span class='op'>*</span> <span class='op'>(</span><span class='op'>-</span><span class='fl'>1</span><span class='op'>)</span><span class='op'>^</span><span class='op'>(</span><span class='va'>i</span> <span class='op'>+</span> <span class='fl'>1</span><span class='op'>)</span> <span class='op'>*</span> <span class='fu'><a href='https://rdrr.io/r/base/Extremes.html'>pmax</a></span><span class='op'>(</span><span class='fl'>1</span> <span class='op'>-</span> <span class='fu'><a href='https://rdrr.io/r/base/MathFun.html'>abs</a></span><span class='op'>(</span><span class='va'>z</span><span class='op'>)</span><span class='op'>^</span><span class='fl'>3</span>, <span class='fl'>0</span><span class='op'>)</span><span class='op'>^</span><span class='fl'>3</span> <span class='co'>## multiply by i to vary size</span>
<span class='op'>}</span>

<span class='co'>## Build data</span>
<span class='va'>beta0</span> <span class='op'>&lt;-</span> <span class='fl'>3</span> <span class='op'>*</span> <span class='fu'><a href='https://rdrr.io/r/base/Trig.html'>sin</a></span><span class='op'>(</span><span class='fl'>2</span> <span class='op'>*</span> <span class='va'>pi</span> <span class='op'>*</span> <span class='va'>pos</span> <span class='op'>/</span> <span class='fl'>720</span><span class='op'>)</span>
<span class='va'>X</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/edgeR/man/cbind.html'>cbind</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/rep.html'>rep</a></span><span class='op'>(</span><span class='fl'>1</span>, <span class='fl'>20</span><span class='op'>)</span>, <span class='fu'><a href='https://rdrr.io/r/base/rep.html'>rep</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>0</span>, <span class='fl'>1</span><span class='op'>)</span>, each <span class='op'>=</span> <span class='fl'>10</span><span class='op'>)</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/Random.html'>set.seed</a></span><span class='op'>(</span><span class='fl'>23852577</span><span class='op'>)</span>
<span class='va'>error</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/matrix.html'>matrix</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/stats/Normal.html'>rnorm</a></span><span class='op'>(</span><span class='fl'>20</span> <span class='op'>*</span> <span class='fu'><a href='https://rdrr.io/r/base/length.html'>length</a></span><span class='op'>(</span><span class='va'>beta1</span><span class='op'>)</span>, <span class='fl'>0</span>, <span class='fl'>1</span><span class='op'>)</span>, ncol <span class='op'>=</span> <span class='fl'>20</span><span class='op'>)</span>
<span class='va'>y</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/t.html'>t</a></span><span class='op'>(</span><span class='va'>X</span><span class='op'>[</span>, <span class='fl'>1</span><span class='op'>]</span><span class='op'>)</span> <span class='op'>%x%</span> <span class='va'>beta0</span> <span class='op'>+</span> <span class='fu'><a href='https://rdrr.io/r/base/t.html'>t</a></span><span class='op'>(</span><span class='va'>X</span><span class='op'>[</span>, <span class='fl'>2</span><span class='op'>]</span><span class='op'>)</span> <span class='op'>%x%</span> <span class='va'>beta1</span> <span class='op'>+</span> <span class='va'>error</span>

<span class='co'>## Perform bumphunting</span>
<span class='va'>tab</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/bumphunter/man/bumphunter.html'>bumphunter</a></span><span class='op'>(</span><span class='va'>y</span>, <span class='va'>X</span>, <span class='va'>chr</span>, <span class='va'>pos</span>, <span class='va'>cl</span>, cutoff <span class='op'>=</span> <span class='fl'>.5</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>[bumphunterEngine] Using a single core (backend: doSEQ, version: 1.5.1).</span></div><div class='output co'>#&gt; <span class='message'>[bumphunterEngine] Computing coefficients.</span></div><div class='output co'>#&gt; <span class='message'>[bumphunterEngine] Finding regions.</span></div><div class='output co'>#&gt; <span class='message'>[bumphunterEngine] Found 15 bumps.</span></div><div class='input'>
<span class='co'>## Explore data</span>
<span class='fu'><a href='https://rdrr.io/r/base/lapply.html'>lapply</a></span><span class='op'>(</span><span class='va'>tab</span>, <span class='va'>head</span><span class='op'>)</span>
</div><div class='output co'>#&gt; $table
#&gt;     chr start  end      value       area cluster indexStart indexEnd  L clusterL
#&gt; 10 chr1  2316 2631 -1.5814747 15.8147473       2         39       48 10       29
#&gt; 7  chr2   451  551  1.5891293  4.7673878       3         68       70  3       20
#&gt; 2  chr1   456  526  1.0678828  3.2036485       1         14       16  3       29
#&gt; 5  chr1  2176 2211  0.7841794  1.5683589       2         35       36  2       29
#&gt; 6  chr1  2841 2841  1.2010184  1.2010184       2         54       54  1       29
#&gt; 4  chr1   771  771  0.7780902  0.7780902       1         23       23  1       29
#&gt; 
#&gt; $coef
#&gt;             [,1]
#&gt; [1,]  0.60960932
#&gt; [2,] -0.09052769
#&gt; [3,] -0.21482638
#&gt; [4,]  0.13053755
#&gt; [5,] -0.21723642
#&gt; [6,]  0.39934961
#&gt; 
#&gt; $fitted
#&gt;             [,1]
#&gt; [1,]  0.60960932
#&gt; [2,] -0.09052769
#&gt; [3,] -0.21482638
#&gt; [4,]  0.13053755
#&gt; [5,] -0.21723642
#&gt; [6,]  0.39934961
#&gt; 
#&gt; $pvaluesMarginal
#&gt; [1] NA
#&gt; </div><div class='input'>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='st'><a href='https://bioconductor.org/packages/GenomicRanges'>"GenomicRanges"</a></span><span class='op'>)</span>

<span class='co'>## Build GRanges with sequence lengths</span>
<span class='va'>regions</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html'>GRanges</a></span><span class='op'>(</span>
    seqnames <span class='op'>=</span> <span class='va'>tab</span><span class='op'>$</span><span class='va'>table</span><span class='op'>$</span><span class='va'>chr</span>,
    <span class='fu'>IRanges</span><span class='op'>(</span>start <span class='op'>=</span> <span class='va'>tab</span><span class='op'>$</span><span class='va'>table</span><span class='op'>$</span><span class='va'>start</span>, end <span class='op'>=</span> <span class='va'>tab</span><span class='op'>$</span><span class='va'>table</span><span class='op'>$</span><span class='va'>end</span><span class='op'>)</span>,
    strand <span class='op'>=</span> <span class='st'>"*"</span>, value <span class='op'>=</span> <span class='va'>tab</span><span class='op'>$</span><span class='va'>table</span><span class='op'>$</span><span class='va'>value</span>, area <span class='op'>=</span> <span class='va'>tab</span><span class='op'>$</span><span class='va'>table</span><span class='op'>$</span><span class='va'>area</span>,
    cluster <span class='op'>=</span> <span class='va'>tab</span><span class='op'>$</span><span class='va'>table</span><span class='op'>$</span><span class='va'>cluster</span>, L <span class='op'>=</span> <span class='va'>tab</span><span class='op'>$</span><span class='va'>table</span><span class='op'>$</span><span class='va'>L</span>, clusterL <span class='op'>=</span> <span class='va'>tab</span><span class='op'>$</span><span class='va'>table</span><span class='op'>$</span><span class='va'>clusterL</span>
<span class='op'>)</span>

<span class='co'>## Assign chr lengths</span>
<span class='fu'>seqlengths</span><span class='op'>(</span><span class='va'>regions</span><span class='op'>)</span> <span class='op'>&lt;-</span> <span class='fu'>seqlengths</span><span class='op'>(</span>
    <span class='fu'>getChromInfoFromUCSC</span><span class='op'>(</span><span class='st'>"hg19"</span>, as.Seqinfo <span class='op'>=</span> <span class='cn'>TRUE</span><span class='op'>)</span>
<span class='op'>)</span><span class='op'>[</span>
    <span class='fu'><a href='https://rdrr.io/r/base/names.html'>names</a></span><span class='op'>(</span><span class='fu'>seqlengths</span><span class='op'>(</span><span class='va'>regions</span><span class='op'>)</span><span class='op'>)</span>
<span class='op'>]</span>

<span class='co'>## Explore the regions</span>
<span class='va'>regions</span>
</div><div class='output co'>#&gt; GRanges object with 15 ranges and 5 metadata columns:
#&gt;        seqnames    ranges strand |     value      area   cluster         L  clusterL
#&gt;           &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;integer&gt;
#&gt;    [1]     chr1 2316-2631      * | -1.581475  15.81475         2        10        29
#&gt;    [2]     chr2   451-551      * |  1.589129   4.76739         3         3        20
#&gt;    [3]     chr1   456-526      * |  1.067883   3.20365         1         3        29
#&gt;    [4]     chr1 2176-2211      * |  0.784179   1.56836         2         2        29
#&gt;    [5]     chr1      2841      * |  1.201018   1.20102         2         1        29
#&gt;    ...      ...       ...    ... .       ...       ...       ...       ...       ...
#&gt;   [11]     chr1       631      * |  0.618603  0.618603         1         1        29
#&gt;   [12]     chr1         1      * |  0.609609  0.609609         1         1        29
#&gt;   [13]     chr1      2911      * | -0.576423  0.576423         2         1        29
#&gt;   [14]     chr2       251      * | -0.556160  0.556160         3         1        20
#&gt;   [15]     chr1      2806      * | -0.521606  0.521606         2         1        29
#&gt;   -------
#&gt;   seqinfo: 2 sequences from an unspecified genome</div><div class='input'>
<span class='co'>## Now create the report</span>
<span class='va'>report</span> <span class='op'>&lt;-</span> <span class='fu'>renderReport</span><span class='op'>(</span><span class='va'>regions</span>, <span class='st'>"Example bumphunter"</span>,
    pvalueVars <span class='op'>=</span> <span class='cn'>NULL</span>,
    densityVars <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span>
        <span class='st'>"Area"</span> <span class='op'>=</span> <span class='st'>"area"</span>, <span class='st'>"Value"</span> <span class='op'>=</span> <span class='st'>"value"</span>,
        <span class='st'>"Cluster Length"</span> <span class='op'>=</span> <span class='st'>"clusterL"</span>
    <span class='op'>)</span>, significantVar <span class='op'>=</span> <span class='cn'>NULL</span>,
    output <span class='op'>=</span> <span class='st'>"bumphunter-example"</span>, outdir <span class='op'>=</span> <span class='st'>"bumphunter-example"</span>,
    device <span class='op'>=</span> <span class='st'>"png"</span>
<span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Writing 11 Bibtex entries ... </span></div><div class='output co'>#&gt; <span class='message'>OK</span>
#&gt; <span class='message'>Results written to file 'bumphunter-example/bumphunter-example.bib'</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>processing file: bumphunter-example.Rmd</span></div><div class='output co'>#&gt;   |                                                                                                                      |                                                                                                              |   0%  |                                                                                                                      |...                                                                                                           |   2%
#&gt;    inline R code fragments
#&gt; 
#&gt;   |                                                                                                                      |.....                                                                                                         |   5%
#&gt; label: docSetup (with options) 
#&gt; List of 3
#&gt;  $ bootstrap.show.code   : logi FALSE
#&gt;  $ dev                   : symbol device
#&gt;  $ bootstrap.show.message: logi FALSE
#&gt; 
#&gt;   |                                                                                                                      |........                                                                                                      |   7%
#&gt;   ordinary text without R code
#&gt; 
#&gt;   |                                                                                                                      |..........                                                                                                    |   9%
#&gt; label: setup (with options) 
#&gt; List of 1
#&gt;  $ bootstrap.show.message: logi FALSE
#&gt; 
#&gt;   |                                                                                                                      |.............                                                                                                 |  12%
#&gt;   ordinary text without R code
#&gt; 
#&gt;   |                                                                                                                      |...............                                                                                               |  14%
#&gt; label: pvaluePlots (with options) 
#&gt; List of 3
#&gt;  $ echo   : logi FALSE
#&gt;  $ results: chr "asis"
#&gt;  $ eval   : symbol hasPvalueVars
#&gt; 
#&gt;   |                                                                                                                      |..................                                                                                            |  16%
#&gt;   ordinary text without R code
#&gt; 
#&gt;   |                                                                                                                      |....................                                                                                          |  19%
#&gt; label: regLen (with options) 
#&gt; List of 4
#&gt;  $ fig.width : num 14
#&gt;  $ fig.height: num 14
#&gt;  $ eval      : symbol hasSignificant
#&gt;  $ echo      : symbol hasSignificant
#&gt; 
#&gt;   |                                                                                                                      |.......................                                                                                       |  21%
#&gt;   ordinary text without R code
#&gt; 
#&gt;   |                                                                                                                      |..........................                                                                                    |  23%
#&gt; label: regLen2 (with options) 
#&gt; List of 4
#&gt;  $ fig.width : num 10
#&gt;  $ fig.height: num 10
#&gt;  $ eval      : language !hasSignificant
#&gt;  $ echo      : language !hasSignificant
#&gt; </div><div class='output co'>#&gt;   |                                                                                                                      |............................                                                                                  |  26%
#&gt;    inline R code fragments
#&gt; 
#&gt;   |                                                                                                                      |...............................                                                                               |  28%
#&gt; label: densityPlots (with options) 
#&gt; List of 3
#&gt;  $ echo   : logi FALSE
#&gt;  $ results: chr "asis"
#&gt;  $ eval   : symbol hasDensityVars
#&gt; </div><div class='output co'>#&gt;   |                                                                                                                      |.................................                                                                             |  30%
#&gt;    inline R code fragments
#&gt; 
#&gt;   |                                                                                                                      |....................................                                                                          |  33%
#&gt; label: genomeOverview1 (with options) 
#&gt; List of 6
#&gt;  $ message   : logi FALSE
#&gt;  $ fig.width : num 7
#&gt;  $ fig.height: num 9
#&gt;  $ dpi       : num 300
#&gt;  $ eval      : symbol hasSignificant
#&gt;  $ echo      : symbol hasSignificant
#&gt; 
#&gt;   |                                                                                                                      |......................................                                                                        |  35%
#&gt;    inline R code fragments
#&gt; 
#&gt;   |                                                                                                                      |.........................................                                                                     |  37%
#&gt; label: manhattanPlots (with options) 
#&gt; List of 3
#&gt;  $ echo   : logi FALSE
#&gt;  $ results: chr "asis"
#&gt;  $ eval   : symbol hasPvalueVars
#&gt; 
#&gt;   |                                                                                                                      |...........................................                                                                   |  40%
#&gt;   ordinary text without R code
#&gt; 
#&gt;   |                                                                                                                      |..............................................                                                                |  42%
#&gt; label: genomeOverview2 (with options) 
#&gt; List of 4
#&gt;  $ message   : logi FALSE
#&gt;  $ fig.width : num 7
#&gt;  $ fig.height: num 9
#&gt;  $ dpi       : num 300
#&gt; </div><div class='output co'>#&gt; <span class='message'>No annotationPackage supplied. Trying org.Hs.eg.db.</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: org.Hs.eg.db</span></div><div class='output co'>#&gt; <span class='message'>Could not load org.Hs.eg.db. Will continue without annotation</span></div><div class='output co'>#&gt; <span class='message'>Getting TSS and TSE.</span></div><div class='output co'>#&gt; <span class='message'>Getting CSS and CSE.</span></div><div class='output co'>#&gt; <span class='message'>Getting exons.</span></div><div class='output co'>#&gt; <span class='message'>Scale for 'x' is already present. Adding another scale for 'x', which will replace the existing scale.</span></div><div class='output co'>#&gt;   |                                                                                                                      |.................................................                                                             |  44%
#&gt;    inline R code fragments
#&gt; 
#&gt;   |                                                                                                                      |...................................................                                                           |  47%
#&gt; label: annoReg (with options) 
#&gt; List of 1
#&gt;  $ results: chr "asis"
#&gt; 
#&gt;   |                                                                                                                      |......................................................                                                        |  49%
#&gt;   ordinary text without R code
#&gt; 
#&gt;   |                                                                                                                      |........................................................                                                      |  51%
#&gt; label: genomeOverview3 (with options) 
#&gt; List of 6
#&gt;  $ message   : logi FALSE
#&gt;  $ fig.width : num 7
#&gt;  $ fig.height: num 9
#&gt;  $ dpi       : num 300
#&gt;  $ eval      : symbol hasSignificant
#&gt;  $ echo      : symbol hasSignificant
#&gt; 
#&gt;   |                                                                                                                      |...........................................................                                                   |  53%
#&gt;    inline R code fragments
#&gt; 
#&gt;   |                                                                                                                      |.............................................................                                                 |  56%
#&gt; label: countTable (with options) 
#&gt; List of 1
#&gt;  $ results: chr "asis"
#&gt; 
#&gt;   |                                                                                                                      |................................................................                                              |  58%
#&gt;   ordinary text without R code
#&gt; 
#&gt;   |                                                                                                                      |...................................................................                                           |  60%
#&gt; label: vennDiagram (with options) 
#&gt; List of 2
#&gt;  $ fig.width : num 7
#&gt;  $ fig.height: num 7
#&gt; </div><div class='output co'>#&gt;   |                                                                                                                      |.....................................................................                                         |  63%
#&gt;    inline R code fragments
#&gt; 
#&gt;   |                                                                                                                      |........................................................................                                      |  65%
#&gt; label: vennDiagramSignificant (with options) 
#&gt; List of 4
#&gt;  $ eval      : symbol hasSignificant
#&gt;  $ echo      : symbol hasSignificant
#&gt;  $ fig.width : num 7
#&gt;  $ fig.height: num 7
#&gt; 
#&gt;   |                                                                                                                      |..........................................................................                                    |  67%
#&gt;    inline R code fragments
#&gt; 
#&gt;   |                                                                                                                      |.............................................................................                                 |  70%
#&gt; label: bestRegionInfo (with options) 
#&gt; List of 1
#&gt;  $ results: chr "asis"
#&gt; 
#&gt;   |                                                                                                                      |...............................................................................                               |  72%
#&gt;   ordinary text without R code
#&gt; 
#&gt;   |                                                                                                                      |..................................................................................                            |  74%
#&gt; label: unnamed-chunk-1 (with options) 
#&gt; List of 2
#&gt;  $ child: symbol customCode
#&gt;  $ eval : symbol hasCustomCode
#&gt; 
#&gt;   |                                                                                                                      |....................................................................................                          |  77%
#&gt;    inline R code fragments
#&gt; 
#&gt;   |                                                                                                                      |.......................................................................................                       |  79%
#&gt; label: thecall (with options) 
#&gt; List of 1
#&gt;  $ echo: logi FALSE
#&gt; 
#&gt;   |                                                                                                                      |..........................................................................................                    |  81%
#&gt;   ordinary text without R code
#&gt; 
#&gt;   |                                                                                                                      |............................................................................................                  |  84%
#&gt; label: reproducibility1 (with options) 
#&gt; List of 1
#&gt;  $ echo: logi FALSE
#&gt; 
#&gt;   |                                                                                                                      |...............................................................................................               |  86%
#&gt;   ordinary text without R code
#&gt; 
#&gt;   |                                                                                                                      |.................................................................................................             |  88%
#&gt; label: reproducibility2 (with options) 
#&gt; List of 1
#&gt;  $ echo: logi FALSE
#&gt; 
#&gt;   |                                                                                                                      |....................................................................................................          |  91%
#&gt;   ordinary text without R code
#&gt; 
#&gt;   |                                                                                                                      |......................................................................................................        |  93%
#&gt; label: reproducibility3 (with options) 
#&gt; List of 1
#&gt;  $ echo: logi FALSE
#&gt; 
#&gt;   |                                                                                                                      |.........................................................................................................     |  95%
#&gt;    inline R code fragments
#&gt; 
#&gt;   |                                                                                                                      |...........................................................................................................   |  98%
#&gt; label: bibliography (with options) 
#&gt; List of 3
#&gt;  $ results: chr "asis"
#&gt;  $ echo   : logi FALSE
#&gt;  $ warning: logi FALSE
#&gt; 
#&gt;   |                                                                                                                      |..............................................................................................................| 100%
#&gt;   ordinary text without R code
#&gt; 
#&gt; </div><div class='output co'>#&gt; <span class='message'>output file: bumphunter-example.knit.md</span></div><div class='output co'>#&gt; /usr/local/bin/pandoc +RTS -K512m -RTS bumphunter-example.utf8.md --to html4 --from markdown+autolink_bare_uris+tex_math_single_backslash --output bumphunter-example.html --lua-filter /__w/_temp/Library/bookdown/rmarkdown/lua/custom-environment.lua --lua-filter /__w/_temp/Library/rmarkdown/rmarkdown/lua/pagebreak.lua --lua-filter /__w/_temp/Library/rmarkdown/rmarkdown/lua/latex-div.lua --self-contained --wrap preserve --standalone --section-divs --table-of-contents --toc-depth 3 --variable toc_float=1 --variable toc_selectors=h1,h2,h3 --variable toc_collapsed=1 --variable toc_smooth_scroll=1 --variable toc_print=1 --template /tmp/Rtmpkhkn6V/BiocStyle/template.html --no-highlight --variable highlightjs=1 --number-sections --css /__w/_temp/Library/BiocStyle/resources/html/bioconductor.css --variable 'theme:bootstrap' --include-in-header /tmp/Rtmpkhkn6V/rmarkdown-str25987905865.html --mathjax --variable 'mathjax-url:https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML' --variable code_folding=hide --variable code_menu=1 </div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Output created: bumphunter-example.html</span></div><div class='input'>
<span class='co'>## See the example report at</span>
<span class='co'>## http://leekgroup.github.io/regionReport/reference/bumphunter-example/bumphunter-example.html</span>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Leonardo Collado-Torres, Andrew E. Jaffe, Jeffrey T. Leek.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


